### PR TITLE
Add email list notification

### PR DESCRIPTION
**Pull request by lzap**
A proposal to create a list people could easily subscribe to:

https://community.theforeman.org/t/how-to-subscribe-to-ci-plugin-failures/25328/6

For now, I am just adding plugins develop tests, but once this is approved I will modify all places so emails are sent for everything.

### CODE REVIEW

File: theforeman.org/yaml/publishers/ircbot_freenode.yaml [hunk 1]
```diff
@@ -29,3 +29,9 @@
             - name: '#theforeman-dev'
               notify-only: true
         matrix-notifier: only-configurations
+
+- publisher:
+    name: email_list
+    publishers:
+    - email:
+        recipients: ci@theforeman.org
```
Comment (ekohl) [inline theforeman.org/yaml/publishers/ircbot_freenode.yaml:37]:
```suggestion
        recipients: ci@community.theforeman.org
```

File: theforeman.org/yaml/publishers/ircbot_freenode.yaml [hunk 2]
```diff
@@ -29,3 +29,9 @@
             - name: '#theforeman-dev'
               notify-only: true
         matrix-notifier: only-configurations
+
+- publisher:
+    name: email_list
```
Comment (ekohl) [inline theforeman.org/yaml/publishers/ircbot_freenode.yaml:34]: I'd prefer to use https://community.theforeman.org/c/infra/21
```suggestion
    name: discourse
```

File: theforeman.org/yaml/jobs/test_plugin.yaml [hunk 3]
```diff
@@ -29,6 +29,7 @@
           block: true
     publishers:
       - ircbot_freenode
+      - email_list
```
Comment (ekohl) [inline theforeman.org/yaml/jobs/test_plugin.yaml:32]: There's no need to modify all jobs. You can override the publishers in the bootdisk plugin config.

File: theforeman.org/yaml/publishers/ircbot_freenode.yaml [hunk 4]
```diff
@@ -29,3 +29,9 @@
             - name: '#theforeman-dev'
               notify-only: true
         matrix-notifier: only-configurations
+
+- publisher:
```
Comment (ekohl) [inline theforeman.org/yaml/publishers/ircbot_freenode.yaml:33]: Please create a new file for this

**Review by ekohl — CHANGES_REQUESTED** [review]

**Comment by lzap** [comment]
Sure, so we have `ci_community.theforeman.org` (scrambled email) for that. Shall I simply create new publisher with this email address?

File: theforeman.org/yaml/jobs/test_plugin.yaml (hunk 3 above)
Comment (lzap) [reply theforeman.org/yaml/jobs/test_plugin.yaml:32]: Sure but woudn't it be more consistent to send all failures there? I think our community could benefit from this, unless it creates too much noise. These are executed weekly, we should be fine.

Comment (ekohl) [reply theforeman.org/yaml/jobs/test_plugin.yaml:32]: I think it creates too much noise and would prefer to start with one. I don't want to end up with a board that has so many things there that nobody looks anymore.

**Comment by ekohl** [comment]
> Sure, so we have `ci_community.theforeman.org` (scrambled email) for that. Shall I simply create new publisher with this email address?

Yes

File: theforeman.org/yaml/jobs/test_plugin.yaml (hunk 3 above)
Comment (lzap) [reply theforeman.org/yaml/jobs/test_plugin.yaml:32]: Let's enable that just for discovery and see how it works.

**Comment by lzap** [comment]
Ok rebased, from the docs it looks like this is how am I supposed to redefine a template, but in the example there they only redefine a variable. I am not sure if this will work for publishers as well. Hopefully we have some tests.

File: theforeman.org/yaml/publishers/discourse.yaml [hunk 5]
```diff
@@ -0,0 +1,5 @@
+- publisher:
+    name: email_list
```
[the remark this answers is not in the document]
Comment (lzap) [reply theforeman.org/yaml/publishers/discourse.yaml:2]: Yeah sorry, renaming stuff and missed it. Fixed.

**Comment by lzap** [comment]
Amended.

**Comment by lzap** [comment]
I guess this was not needed. Thanks!

**Pull request closed without merging**